Changes in sky cultures

alex-w opened the pull request:
### Description
This pull request add the support for parameter "region" for sky cultures. This parameter can be using as informational data for users and also for developers to prepare downloadable packs of sky cultures (the support of SC packages should be added in separate PR).

### Type of change
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] This change requires a documentation update

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the [code style](http://stellarium.org/doc/head/codingStyle.html) of this project.
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (header file)
- [ ] I have updated the respective chapter in the Stellarium User Guide
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

alex-w commented:
I think the PR is ready to review & merge

alex-w commented:
@gzotti @sushoff please review the PR

gzotti commented:
I am not sure I like the renaming away from "Western". But it makes the names of the derivatives, Western_Rey, Western_SnT, etc., less clear. 

And BTW, those last two are certainly not region=Western Europe: S&T is USA, and Rey's book is also most popular there.  So both could be "North America", but we intend to show indigenous North American SCs under this group. I think for all SCs from the tradition of Western science we should use the pseudo-region "Western" (without continent) to mark "tradition of Western science". In this sense, the current Western should remain that.  And yes, also an astronomical atlas published by the Soviet Academy of Sciences in Moscow in the 1950s would fall under this category, unless it shows constellations of Marx and Lenin or similar disruptions.  

What I agree upon is the complete rewrite of the text. I have added a few things which I think are important to say w.r.t. our displays.

alex-w commented on skycultures/world/description.en.utf8:
```diff
@@ -51,24 +51,42 @@ thus, was common in the Near East, in Europe and its colonies on all continents.
 <p>Stellarium by default displays the precise RA-DEC-boundaries and a set of artwork constellations on the basis of Ptolemaic figures but following our contemporary taste of art.</p>
 
+<h3>Asterisms</h3>
+The term "asterism" (assembly of stars) can sometimes be used synonymously to "constellation". In the modern use it describes a figure which is not part of the official canon. 
```
Thanks for additional, but please define paragraphs (<p></p>) for text of these additionals

gzotti replied:
Oops, yes!

alex-w commented:
> And BTW, those last two are certainly not region=Western Europe: S&T is USA, and Rey's book is also most popular there. So both could be "North America", but we intend to show indigenous North American SCs under this group. I think for all SCs from the tradition of Western science we should use the pseudo-region "Western" (without continent) to mark "tradition of Western science". In this sense, the current Western should remain that. And yes, also an astronomical atlas published by the Soviet Academy of Sciences in Moscow in the 1950s would fall under this category, unless it shows constellations of Marx and Lenin or similar disruptions.

Well, this is acceptable and probably we should add a short explanation for the pseudo-region "Western" - could you write some text for it?

alex-w commented:
@gzotti please check the description of Western pseudo-region

sushoff commented:
I do like the change "Western" to "world" - LOVE IT! 
I would rather also change "world_S&T, "world_O.Hlad" and "World_Rey" because they are all derivatives within the borders of IAU-definition. Please let's change it back to the original change. In the IAU, there seems to be the consent that we want to avoid the term "western" in the future but a) want to be more inclusive and b) it doesn't make sense on a spherical Earth. 

I would also change "greek_Almagest" to "Northern Africa" because "greek" is the language (academic lingua franca) and not the "Greek" that we think today as "in the country of today's Greece". 

I always wonder why we don't write "Australia" but "Australasia". Culturally, that appears strange because peoples of Oceania had different cultures than those on the continent. I loved your solution, Alex, to use continents for "regions" instead of countries. But might it be better to use directly the geological continents (although we might run into little funnies when "Sinai" becames and own region this way, neither belonging to Africa nor to Eurasia ... but that fits the culture, I think).

alex-w commented:
> I do like the change "Western" to "world" - LOVE IT! I would rather also change "world_S&T, "world_O.Hlad" and "World_Rey" because they are all derivatives within the borders of IAU-definition. Please let's change it back to the original change. In the IAU, there seems to be the consent that we want to avoid the term "western" in the future but a) want to be more inclusive and b) it doesn't make sense on a spherical Earth.

@gzotti is it acceptable for you?

> I would also change "greek_Almagest" to "Northern Africa" because "greek" is the language (academic lingua franca) and not the "Greek" that we think today as "in the country of today's Greece".

OK, this is good remark

> I always wonder why we don't write "Australia" but "Australasia". Culturally, that appears strange because peoples of Oceania had different cultures than those on the continent. I loved your solution, Alex, to use continents for "regions" instead of countries. But might it be better to use directly the geological continents (although we might run into little funnies when "Sinai" becames and own region this way, neither belonging to Africa nor to Eurasia ... but that fits the culture, I think).

We use United Nations geoscheme (UN M49) for locations already and I re-use it again.

alex-w commented:
> Or what about naming it "IAU_ etc "

I disagree for using this naming, because we should follow strict rules from IAU for this case and we should remove all "western" SC, because all of them is not well-formed.

sushoff commented:
> We use United Nations geoscheme (UN M49) for locations already and I re-use it again.

ah, ok ... that's acceptable, good idea (I remember now, was Georg's suggestion) 

personally, I would try to avoid political borders and find an approach that is based in nature - but the UN is probably the most neutral ground in which we can plant our plants

sushoff commented:
> > Or what about naming it "IAU_ etc "
> 
> I disagree for using this naming, because we should follow strict rules from IAU for this case and we should remove all "western" SC, because all of them is not well-formed.

oh?! oh, I did not know that they were not well-formed. 

But I see your point: the IAU as an institution doesn't is not involved in the software development. "Delporte" would be too specific as name for the SC

So, your initial term "world" (or modern or sth. like this) is proberbly the best.

gzotti commented:
The difficulties in the definition are pretty well described in https://en.wikipedia.org/wiki/Western_world

I just added a sentence how we can understand this term. Scientific, not political. 

Every scheme has its problems. The UNO Geoscheme dissects into continents, AFAIK along borders of today's countries (Uh, what about Turkey?). Yes, Ptolemy lived in Alexandria, but was part of the Greek/Hellenistic world living in a Roman colony. By culture, he can likely be described as Greek, Mediterranean or even ... simply Western, as he is one early and important representative.

sushoff commented:
PS: The IAU does not have any other definition as the boundaries (and they are correctly taken from the IAU AFAIK). The IAU does not given (and has decided to never will give) rules for a) art work, b) boundaries, c) pronunciations ... AFAIK. at least that is written on their website (that I have recently revisited intensively with many other groups)

sushoff commented:
> The difficulties in the definition are pretty well described in https://en.wikipedia.org/wiki/Western_world
> 
> I just added a sentence how we can understand this term. Scientific, not political.
> 
> Every scheme has its problems. The UNO Geoscheme dissects into continents, AFAIK along borders of today's countries (Uh, what about Turkey?). Yes, Ptolemy lived in Alexandria, but was part of the Greek/Hellenistic world living in a Roman colony. By culture, he can likely be described as Greek, Mediterranean or even ... simply Western, as he is one early and important representative.

the nonsense had been lasting long - so let's continue it... what a mess! 

Can we please simply change it to the better?!

alex-w commented:
> PS: The IAU does not have any other definition as the boundaries (and they are correctly taken from the IAU AFAIK). The IAU does not given (and has decided to never will give) rules for a) art work, b) boundaries, c) pronunciations ... AFAIK. at least that is written on their website (that I have recently revisited intensively with many other groups)

Yes, this is correct statement

gzotti commented:
We can have (and I would welcome this!) a "Western_IAU" SC which is strictly limited to 
- star names spelling from the WGSN. No other names. 
- 88 IAU constellations
- no asterisms, no ray helpers
- no artwork (?)
- stick figures: S&T has published a set for IYA2009 which was allegedly developed in collaboration with IAU experts. We could probably use that. Or is that already in Western_SnT? I lost track. :-)


Our default Western SC is a collection from tens of books of starnames from the "Western" scientific tradition. And just that. No sky guide from Africa, Oceania or China has been sourced for star names.  Yes, the IAU names should also be in there, but one of my aims in Skycultures2.0 will be to be able select the used source material (references).

sushoff commented:
> We can have (and I would welcome this!) a "Western_IAU" SC which is strictly limited to
> 
> * star names spelling from the WGSN. No other names.
> * 88 IAU constellations
> * no asterisms, no ray helpers
> * no artwork (?)
> * stick figures: S&T has published a set for IYA2009 which was allegedly developed in collaboration with IAU experts. We could probably use that. Or is that already in Western_SnT? I lost track. :-)
> 
> Our default Western SC is a collection from tens of books of starnames from the "Western" scientific tradition. And just that. No sky guide from Africa, Oceania or China has been sourced for star names. Yes, the IAU names should also be in there, but one of my aims in Skycultures2.0 will be to be able select the used source material (references).

I know that this SC is a mixture and I welcome this. My problem is the term "Western". First, because it does not make any sense on a globe and is (historically/ politically spoken Eurocentric - and cultural research wishes to overcome Eurocentrism since the 1970s). Second, because it reminds on and emphasizes a political division of the world that we want to overcome since the 1990s. 

The first issue is probably more important. I really don't care for what reasons earlier scholars had been invented some decades ago: If these terms are not appropriate anymore, I want to change them. "western" is one of them. I strongly dislike this term - the same (literally same) amount as I dislike the "lingua tertii emperii".

sushoff commented:
- Greek_Almagest should be Northern Africa (because it was written in Alexandria) 
- Greek_Leiden should be Western Europe (because the MS was found in France and NL)
- Greek_Farnese is correct (Southern Europe)

Atque commented:
> * Greek_Almagest should be Northern Africa (because it was written in Alexandria)
> * Greek_Leiden should be Western Europe (because the MS was found in France and NL)
> * Greek_Farnese is correct (Southern Europe)

Shouldn't the location of the culture be taken into account? Greek culture = Southern Europe. Otherwise, it would mean that a book about an Australian Aboriginal sky culture, written in the UK, would be classified as European.

gzotti commented:
> > I do like the change "Western" to "world" - LOVE IT! 

I knew you would say that...

>I would rather also change "world_S&T, "world_O.Hlad" and "World_Rey" because they are all derivatives within the borders of IAU-definition. Please let's change it back to the original change. In the IAU, there seems to be the consent that we want to avoid the term "western" in the future but a) want to be more inclusive and b) it doesn't make sense on a spherical Earth.
> 
> @gzotti is it acceptable for you?

As said, this "western" is not geographic, but a term which describes the prevalent way of doing science, which happened to have evolved in what for centuries has been known as the Western world which BTW nowadays also includes far-Eastern regions like Australia. Yes, it is confusing, but in the same way ubiquitously understood. 

If the IAU had, 100 years ago, included constellations from other cultural regions, then yes, this would justify a term like "World" SC. Adding a few indigenous star names to stars firmly placed inside the borders of Western constellations is like attaching a colorful bow to a gray cuboid cardboard box. The box remains a gray cuboid and will not grow wings.  We simply have no "World" SC that could seriously bear this name, sorry. 

It would be a funny enterprise to create one inofficial "World SC" for now: Take the best-known figures from around the world. Concentrate on non-European traditions. Constellations, no borders. Allow overlap. Allow multiple assignments for super-conspicuous figures. Artwork and stick figures. Dark constellations.  This would be a wonderful "best of the world" show.

gzotti commented:
> PS: The IAU does not have any other definition as the boundaries (and they are correctly taken from the IAU AFAIK). The IAU does not given (and has decided to never will give) rules for a) art work, b) boundaries, c) pronunciations ... AFAIK. at least that is written on their website (that I have recently revisited intensively with many other groups)

(b) recte: stick figures?

sushoff commented:
> > * Greek_Almagest should be Northern Africa (because it was written in Alexandria)
> > * Greek_Leiden should be Western Europe (because the MS was found in France and NL)
> > * Greek_Farnese is correct (Southern Europe)
> 
> Shouldn't the location of the culture be taken into account? Greek culture = Southern Europe. Otherwise, it would mean that a book about an Australian Aboriginal sky culture, written in the UK, would be classified as European.

no: The Leiden MS shows partially Christianized Roman versions of the Aratos-constellations - i.e. it is rooted in the Greek tradition but 1.5 millennium later - dating to the Holy Roman Empire. It is a derivative (like SnT and Rey of the "World" culture). 

same with Farnese: shows a specific interpretation. The marble globe dates to +2nd century (time of the Almagest) but shows different constellations than the Almagest, however, rooting in the same tradition that we call "Greek" 

but the clue is: they are all different: That's why I defined them as separate distinguished SCs

sushoff commented:
> > PS: The IAU does not have any other definition as the boundaries (and they are correctly taken from the IAU AFAIK). The IAU does not given (and has decided to never will give) rules for a) art work, b) boundaries, c) pronunciations ... AFAIK. at least that is written on their website (that I have recently revisited intensively with many other groups)
> 
> (b) recte: stick figures?

yes, thank you

gzotti commented:
I think a rename needs more work with some experts, maybe there is still a better term, I just cannot find it. I should be on my way to the right people just now. With some luck, I hope to fly next week. Now I should rest.

sushoff commented:
New suggestion for "Western": let's just rename them 

-  "Default (Stellarium)"  
- Default_Rey 
- Default_SnT
- Default_OHlad

alex-w commented:
OK, renaming can be done in separate PR - what about region support?

gzotti commented:
OK, in principle, as it will enable us to create regional packages. But what about the "Western" pseudo-region, under which I would collect everything that roots in Almagest? (Even Sufi? This work is a "fork", on the one side collecting Almagest+Arab traditions back into later Maghreb editions of Almagest and by this, back to Europe. But it also spread Almagest into the Islamic science, e.g., Maragha, Samarqand, so this regionalisation is not easy. Also note the European interest in and print of Ulug-Bek's tables which stand in Almagest tradition but are usually regarded as late outcrops of "Islamic science") If we plan to rename what the world has known and has been using under the term "Western" for at least the past few centuries, then it should be OK to withhold and discuss this change for a few more weeks. I am waiting for suggestions from social/cultural experts for a new term which encompasses everything that so far has been understood under this term. 

I don't like "Default" as this just means "Vorgabe". It is up to anyone to store one SC as personal default.  "World" implies global input of which there is certainly not enough in the current "Western" SCs. "Scientific mainstream"? weird kludge. But even Chinese nuclear physicists most likely build their machinery on basis of "Western" physics (filtered by Soviet physicists), not Feng-Shui or the Mao scripts. 

What if this pseudo-region of "scientific mainstream astronomy" is not themed "Western" but "Ptolemaic"? This would encompass all of the above when rooted in Almagest. (Even Schiller, as he just reinterpreted existing figures.) What is NOT in here are Arab mansions, European SCs like Sardinia, Romanian, Belarus (although there are again overlaps...) The Aratea? If mixed with Almagest, it should be included. Else it's Mediterranean/S.Europe (I tend to agree with @Atque here.) We could also have a temporal split in the packages. "Ptolemaic pre-1500" (this could also include "Islamic" items up to Ulug-Bek), "Ptolemaic Renaissance (1501-1850 [1880?])" (This to include the Dürer planispheres of 1515 up to the great copperprint atlases until Bode, and 19th century classics like Littrow or Mädler), "Ptolemaic Modern (1851[1881?]-now)" (including mostly modern scientific partitions, but also late popularisations like Burritt; things that can be occasionally found in own family heritage or used-book stores (not special auction stuff). The 1850/1880 is arbitrary and could be defined more clearly. E.g. AG [sorry, another mainstream Western society...] was registered in 1863...).

The main package would then only need to include the "Ptolemaic Modern" package, and interested users can download more. The remaining issue will be finding a name that Susanne likes for what is now called Western, Western_SnT, Western_Rey, Western_hlad. I would not call them Ptolemaic_* directly as they have evolved too far into the `<unspeakable>` scientific tradition. Maybe "`Modern_*`"? This would be good IMHO to express "Actively used by amateurs and professionals worldwide today". Then we need the `Modern_IAU` variant as described above, strictly no other star names beyond the IAU WGSN and Exoplanets lists.  Then we should agree on a collective replacement term for possible future "Western_Hevelius", "Western_Bayer", "Western_Bode", "Western_Schiller", .... Maybe `Classic_*`? 

This is just input to a thought process. I see no need to rush it. It may be better to plan for 1.2. (Also to allow >2 weeks for translation!)

alex-w commented:
`Modern_*` for all SC with IAU's official boundaries (e.g. all "Western" since 1930) and `Classic_*` for all "Western" until 1930. Probably region shouldn't be defined for all Modern_* SC.

gzotti commented:
Also possible. I would then just not use 1930 but the difference of "IAU borders" vs. "other borders". I.e. `Modern_*` must use Delporte/IAU borders. (A hypothetical edition of an atlas printed 1932 with curved sky regions or even non-standard constellations would not classify as "modern".) Older Ptolemaic after 1500 are then `Classic_*`. Older ones likely directly `Almagest_*` (where applicable). And for region I would then (as "Renaissance" should probably really not extend into the 20th century...) use 
- `Ptolemaic pre-1500`
- `Ptolemaic Classic`
- `Ptolemaic Modern`

The first name can be improved. `Ptolemaic Historical`? (`Ptolemaic Antiquity and Medieval` is too long IMHO.)

Now please please don't object that "Ptolemaic" means Hellenistic Greek... Yes, in yet another context!

alex-w commented:
> Also possible. I would then just not use 1930 but the difference of "IAU borders" vs. "other borders". I.e. `Modern_*` must use Delporte/IAU borders. (A hypothetical edition of an atlas printed 1932 with curved sky regions or even non-standard constellations would not classify as "modern".) 

+1 for "I.e. `Modern_*` must use Delporte/IAU borders" - this is was I meant. I guess I saw Atlas from 1935 with curve borders.

> Older Ptolemaic after 1500 are then `Classic_*`. Older ones likely directly `Almagest_*` (where applicable). And for region I would then (as "Renaissance" should probably really not extend into the 20th century...) use
> 
> * `Ptolemaic pre-1500`
> * `Ptolemaic Classic`
> * `Ptolemaic Modern`
> 
> The first name can be improved. `Ptolemaic Historical`? (`Ptolemaic Antiquity and Medieval` is too long IMHO.)

`Archaic`?

gzotti commented:
> `Archaic`?

No, that predates classical Antiquity. I need a term that should express "Antiquity+Medieval". Given that new constellations were added only after 1500, we could name this whole early group also simply `Almagest`. That means, we would have the UNO M49 continental regions plus three pseudo-regions, 
- Almagest
- Ptolemaic Classic
- Ptolemaic Modern

or even just

- Almagest
- Classic
- Modern

but using "Classic" for European Renaissance/premodern Almagest derivatives is again a eurocentric view, given that there is also "Maya Classic" and "Modern Chinese". So I would prefer the first version.

sushoff commented:
> Also possible. I would then just not use 1930 but the difference of "IAU borders" vs. "other borders". I.e. `Modern_*` must use Delporte/IAU borders. (A hypothetical edition of an atlas printed 1932 with curved sky regions or even non-standard constellations would not classify as "modern".) Older Ptolemaic after 1500 are then `Classic_*`. Older ones likely directly `Almagest_*` (where applicable). And for region I would then (as "Renaissance" should probably really not extend into the 20th century...) use
> 
> * `Ptolemaic pre-1500`
> * `Ptolemaic Classic`
> * `Ptolemaic Modern`
> 
> The first name can be improved. `Ptolemaic Historical`? (`Ptolemaic Antiquity and Medieval` is too long IMHO.)
> 
> Now please please don't object that "Ptolemaic" means Hellenistic Greek... Yes, in yet another context!

sushoff commented:
Thank you, Georg for all these thoughts. 

It was because of the IAU-borders (that are clearly named "IAU" in the description files) that I suggested this term. But I agree with your arguments that everything else but the borders is not "Iau"-defined. 

I would not call it "Ptolemaic" because this will cause trouble for ancient history (e.g. Farnese is not "clearly"/ only Ptolemaic ...) and it will appear strange to a user, I guess, to not combine them in a comparative way. 

I like Alexander's new suggestion 
-  modern_* for the "western" after Delporte 
- classic_* for everything before 1930 

I was thinking a moment if the term "classic" could be misunderstood by classicists - but I think: no, it will be alright, I think.

gzotti commented:
Yes the Farnese SC should not be in the Ptolemaic/Almagest group when it contains the Aratean sky. This would rather be South Europe. A mixed Farnese/Almagest SC could however be included. 

But if we can agree on a "classic" pseudo-region for [all that was previously called "western" except for [everything previously termed "western" that uses IAU borders which goes into a new "modern" pseudo-region]], we can do that. In this case, I would then place Farnese and (Leiden and other editions of) Aratea into the "classic" group. This "classic" would basically include all depictions from classical Graeco-Roman antiquity, European Medieval, and European Renaissance and "Age of Discoveries" until before the IAU formalisation.

alex-w commented:
> But if we can agree on a "classic" pseudo-region for [all that was previously called "western" except for [everything previously termed "western" that uses IAU borders which goes into a new "modern" pseudo-region]], we can do that. In this case, I would then place Farnese and (Leiden and other editions of) Aratea into the "classic" group. This "classic" would basically include all depictions from classical Graeco-Roman antiquity, European Medieval, and European Renaissance and "Age of Discoveries" until before the IAU formalisation.

No need pseudo-regions. All `modern_*` SC shouldn't have region option by definition. All `classic_*` SC should have define region from region of origin (if it possible).

sushoff commented:
so we would have 

- classic_Almagest
- classic_Alm+Farnese (? not South. Eur.?)
- classic_Alm+Leiden  (? not Western Eur.?) 
- classic_As-Sufi (????) 
- classic_Bode / Bayer / Schiller ... whoever will be added 

and 

- modern (Stellarium) 
- modern (Hlad)
- modern (Rey) 
- modern (SnT) 

Or should we leave out the "classic" first and let the three existing "Greek" and the other "Arabic" SCs? 

I would be fine if we started - for the time being - by renaming "western" to sth else.

sushoff commented:
additional thought (just for beauty) 

Wouldn't it be user-friendly to have the "Stellarium home culture" (whatever we decide to call it) at a prominent place in the display, i.e. at the beginning or end of the list of all SCs? we could add a "0" to put it always on top.

gzotti commented:
> additional thought (just for beauty)
> 
> Wouldn't it be user-friendly to have the "Stellarium home culture" (whatever we decide to call it) at a prominent place in the display, i.e. at the beginning or end of the list of all SCs? we could add a "0" to put it always on top.

Probably yes, we may put all modern_* on top where most users would expect them. Do you mind if this gives prevalence to some over others? Technically, I think I have just seen a specialized sort in the search panel, this idea can also work here.

gzotti commented:
> > But if we can agree on a "classic" pseudo-region for [all that was previously called "western" except for [everything previously termed "western" that uses IAU borders which goes into a new "modern" pseudo-region]], we can do that. In this case, I would then place Farnese and (Leiden and other editions of) Aratea into the "classic" group. This "classic" would basically include all depictions from classical Graeco-Roman antiquity, European Medieval, and European Renaissance and "Age of Discoveries" until before the IAU formalisation.
> 
> No need pseudo-regions. All `modern_*` SC shouldn't have region option by definition. All `classic_*` SC should have define region from region of origin (if it possible).

Hmm, indeed, you have discussed away the Almagest pedigree.

gzotti commented:
> so we would have
> 
>     * classic_Almagest
>     * classic_Alm+Farnese (?  South. Eur.?)
>     * classic_Alm+Leiden  (? Western Eur.?) 
>     * classic_As-Sufi (????)
>     * classic_Bode / Bayer / Schiller ... whoever will be added
> 
> 
> and
> 
>     * modern (Stellarium)
>     * modern (Hlad) 
>     * modern (Rey)
>     * modern (SnT)
> 
> 
> Or should we leave out the "classic" first and let the three existing "Greek" and the other "Arabic" SCs?

The "classic_*" IMHO helps to group the early mainstream/whicheveryounotwanttocallthem SCs in the alphabetized list. We could think of yet another group name for those which came before my "classic" group which was the 1500-1850 (then 1920) era of "classic copperprint figure atlases".

alex-w commented:
Important note: I renamed all "western" SC into "modern" SC and our default (the "Stellarium home culture") SC is named simple "Modern". I think we should add a strict "Modern (IAU)" [SCID: `modern_iau`] SC with IAU-approved boundaries and star names. Probably without artwork, but with stick figures from S&T SC.

For SC with SCID `modern_*` I added pseudo-region `World` with short explanation/description `All modern sky cultures based on approved by IAU 88 constellations with standardized boundaries and used onto whole world. The origins all of these constellations are pan-European.`

I think we may add additions about "the Almagest pedigree" to the description of region for SC with SCID `classic_*` when the `region` is defined and adding pseudo-region `Unknown` with explanation about difficulty of determination the origin region, but the SC is based on (is derivative of) the Almagest.

gzotti commented:
> Important note: I renamed all "western" SC into "modern" SC and our default (the "Stellarium home culture") SC is named simple "Modern". I think we should add a strict "Modern (IAU)" [SCID: `modern_iau`] SC with IAU-approved boundaries and star names. Probably without artwork, but with stick figures from S&T SC.

+1. It's a "single" presentation. No asterisms, no ray helpers. Presumably also no fancy DSO names! (Do we "allow" common, well-established names for Messier and the Caldwell selection for the benefit of contemporary amateur astronomers/observers? What list do we allow? Burnham? Sky Catalog 2000.0?)

> For SC with SCID `modern_*` I added pseudo-region `World` with short explanation/description `All modern sky cultures based on approved by IAU 88 constellations with standardized boundaries and used onto whole world. The origins all of these constellations are pan-European.`

OK for me

> I think we may add additions about "the Almagest pedigree" to the description of region for SC with SCID `classic_*` when the `region` is defined and adding pseudo-region `Unknown` with explanation about difficulty of determination the origin region, but the SC is based on (is derivative of) the Almagest.

If it is Almagest+regional additions, it can be localized now. Just the names "classic_*" help in visual grouping as long as we don't develop a multi-list GUI with all kinds of filters and selections.  I would like to avoid an "unknown" region. But we must remain flexible if something unexpected turns up.

gzotti commented:
> > 
> > The "classic_*" IMHO helps to group the early mainstream/whicheveryounotwanttocallthem SCs in the alphabetized list. We could think of yet another group name for those which came before my "classic" group which was the 1500-1850 (then 1920) era of "classic copperprint figure atlases".
> 
> mhhhh ... difficult: the grouping in epochs is not really easy and many cultures cannot really be dated properly: in particular, all the ethnological ones: probably they also go back to stoneage but have been changed continuously in oral traditions due to precession and other effects ...

I just meant grouping so that they are better visible in the alphabetized list. We can create probably tens of classic-Almagest derived SCs. (At least in principle. Cleaning up scans from only one atlas takes weeks!)  Most of these works have been published in Europe. Still, they should stand out over the non-Almagest European SCs.  This is the big exception. The only other cultures that could be presented in several related SCs starting with some common name part are Chinese and descendents (Korea, Japan, ... ; But I am afraid visual grouping in this case will lead to political tensions in this crazy world!), and if known well enough, pre- and Islamic Arab SCs.  For the others I see not more than one description by some ethnologists, so these are free to use other names that should just not start with "classic_".  


> 
> however, I guess, groupings would help
> 
> I'd prefer a grouping with regard to continents (and here I literally mean the continents in sense of geography) because it might help planetarium educators and teachers to find their local indigenous/ historical and neighbour culture(s).

At some point we must add a region selector to the GUI. Either a filter that allows this and that into the current selection list, or we even must replace the current list and present installed SCs in a graph-like structure on one panel (maybe overlaying a world map...), and selecting one graph node will then load it. Phew, who is going to develop that? It does not fit to my current activities, I am afraid. One problem surely is the question what to map? The UN scheme gives a structure by-continent. We could probably add "Ptolemaica" as island to Europe. Another graph could try to map temporal development. But this could (for the few rare cases where we can trace a temporal development) also be mapped to a "street map" inside "Ptolemaica". As said, we will probably not be able to decide upon these points this week.

alex-w commented:
> > Important note: I renamed all "western" SC into "modern" SC and our default (the "Stellarium home culture") SC is named simple "Modern". I think we should add a strict "Modern (IAU)" [SCID: `modern_iau`] SC with IAU-approved boundaries and star names. Probably without artwork, but with stick figures from S&T SC.
> 
> +1. It's a "single" presentation. No asterisms, no ray helpers. Presumably also no fancy DSO names! (Do we "allow" common, well-established names for Messier and the Caldwell selection for the benefit of contemporary amateur astronomers/observers? What list do we allow? Burnham? Sky Catalog 2000.0?)

Probably well-established names for objects from Messier catalog only - see https://iau.org/public/themes/constellations/ 

> > I think we may add additions about "the Almagest pedigree" to the description of region for SC with SCID `classic_*` when the `region` is defined and adding pseudo-region `Unknown` with explanation about difficulty of determination the origin region, but the SC is based on (is derivative of) the Almagest.
> 
> If it is Almagest+regional additions, it can be localized now. Just the names "classic_*" help in visual grouping as long as we don't develop a multi-list GUI with all kinds of filters and selections. I would like to avoid an "unknown" region. But we must remain flexible if something unexpected turns up.

Currently users will see nothing if `region` is not defined. So, no need extra pseudo-region and additions to the description.

alex-w commented:
> At some point we must add a region selector to the GUI. Either a filter that allows this and that into the current selection list, or we even must replace the current list and present installed SCs in a graph-like structure on one panel (maybe overlaying a world map...), and selecting one graph node will then load it. Phew, who is going to develop that? It does not fit to my current activities, I am afraid. One problem surely is the question what to map? The UN scheme gives a structure by-continent. We could probably add "Ptolemaica" as island to Europe. Another graph could try to map temporal development. But this could (for the few rare cases where we can trace a temporal development) also be mapped to a "street map" inside "Ptolemaica". As said, we will probably not be able to decide upon these points this week.

Sorry, but all of these questions are not aim of current PR.

gzotti commented:
The current state of the PR is still with "western". Please first rename to modern_*. We shall also then finalize the description to take out the terminology undisputed in the former two decades.

alex-w commented:
> The current state of the PR is still with "western". Please first rename to modern_*. We shall also then finalize the description to take out the terminology undisputed in the former two decades.

Where is the old name?

gzotti commented on skycultures/modern/description.en.utf8:
```diff
@@ -0,0 +1,91 @@
+thus, was common in the Near East, in Europe and its colonies on all continents. 
+However, between the ancient Greek constellations there were gaps of areas with only faint stars belonging to no constellation.[#47] 
+In the Modern Epoch, several astronomers suggested constellation figures for these gaps 
+(e.g. the constellation of "the lynx" for an area close to the Great Bear but with stars so faint that you would need the eyes of a lynx to see them), 
```
shall we really have this sentence twice in the text? @sushoff ?

sushoff replied:
? what do you mean?

gzotti replied:
the line about the lynx constellation being so dim that you need lynx eyes to see it can be read twice within this text. There may be other redundant bits, but other things have distracted my attention these days.

gzotti commented on src/core/StelSkyCultureMgr.cpp:
```diff
@@ -279,6 +282,25 @@ QString StelSkyCultureMgr::getCurrentSkyCultureHtmlLicense() const
+	{
+		// TRANSLATIONS: By the fact this is name of pseudo-region on Earth
+		region = N_("World");
+		description = q_("All modern sky cultures based on approved by IAU 88 constellations with standardized boundaries and used onto whole world. The origins all of these constellations are pan-European.");
```
All modern sky cultures are based on the IAU-approved 88 constellations with standardized boundaries and are used worldwide. The origins of all these constellations are pan-European.

gzotti commented on src/scripting/StelMainScriptAPI.cpp:
```diff
@@ -427,7 +427,10 @@ QString StelMainScriptAPI::getSkyCulture()
 {
 	GETSTELMODULE(StelObjectMgr)->unSelect(); // mistake-proofing!
-	emit requestSetSkyCulture(id);
+	if (id=="western") // for backward compatibility of scripts
```
better use id.startsWith("western")

alex-w replied:
I disagree, because we should run old scripts for the default SC (SCID: western) in the new environment.

gzotti replied:
I did not check further. Does the request...(id) give an error message if western_hlad is not found? Or some other recovery? The startsWith would cause a fallback to the current modern. 
Probably also here should be a note to users: 

`qDebug() << "Western skycultures have been renamed to modern_*. Please update your scripts."`

alex-w replied:
OK, just informing about changes and do nothing in script...

alex-w commented:
@gzotti @sushoff is description of modern SC OK?
